### PR TITLE
Adds more sections of developing instructions in README.md

### PR DESCRIPTION
**Pull request by linlin-s**
Descriptions:
Add more sections of instructions in README.md about adding more options to Ion Data Generator and extending the supported data type of Ion Data Generator. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,6 +202,38 @@ the tool by hand to make sure everything looks correct and the benchmark results
...
+1. In `Main`, add the option to the `USAGE` and `OPTIONS` strings, mimicking the existing format.
+2. Add invalid options hashset to `GeneratorOptionsValidator` , mimicking the existing format.
+3. Add parsing options logic to `GeneratorOptions` and consider which type of Ion data the parsed parameters should be passed to, 
+   mimicking the existing logic of passing options parameters.
+4. In `WriteRandomIonValues`, add logic to the methods of constructing Ion Data to process the added option.
+
+### Adding more supported data type to Ion Data Generator
+
+Adding abilities to Ion Data Generator to generate more types of Ion Data which conform with the Ion Schema.
+1. In `ReadGeneralConstaints`, add more data type cases under switch logic, mimicking the existing supported data type.
```
Comment (tgregg) [inline README.md:217]: `ReadGeneralConstaints` -> `ReadGeneralConstraints`

File: README.md [hunk 2]
```diff
@@ -202,6 +202,38 @@ the tool by hand to make sure everything looks correct and the benchmark results
...
+2. Add methods in `WriteRandomIonValues`, mimicking the existing methods which writes random ion data to the generated file. 
+   If the data type is already supported by the existing methods, find the relative method to see if more logic processing the constraints needed to be added.
```
Comment (tgregg) [inline README.md:219]: relative -> related

**Review by tgregg — APPROVED** [review]
Thanks!

File: README.md (hunk 1 above)
Comment (linlin-s) [reply README.md:217]: > `ReadGeneralConstaints` -> `ReadGeneralConstraints`

Changes will be updated in the next commit.

File: README.md (hunk 2 above)
Comment (linlin-s) [reply README.md:219]: > relative -> related

Changes will be updated in the next commit.

**Review by linlin-s — COMMENTED** [review]
The typos have been fixed in the second commit.

**Pull request merged**